### PR TITLE
Complete Rebuild of gem structure, usage interface

### PR DESCRIPTION
**Pull request by DmitryDrobotov**
closes https://github.com/riboseinc/transcryptor/issues/11

### CODE REVIEW

**Comment by ronaldtse** [comment]
Thanks @DmitryDrobotov ! There are a couple typos in the README, but this structure itself is a truly a great improvement.

File: spec/transcryptor/instance_spec.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    InstanceSpec.all.each_with_index do |instance_spec, i|
+      expect(instance_spec.column_1).to eq("value_#{i}")
+    end
+    InstanceSpec.instance_variable_set(:@encrypted_attributes, {})
+  end
+
+  describe '#re_encrypt' do
+    context 'from per_attribute_iv to per_attribute_iv_and_salt' do
+      let(:attr_encrypted_config_before) do
+        InstanceSpec.send(
```
Comment (ronaldtse) [inline spec/transcryptor/instance_spec.rb]: Is it necessary to use `send(` here or just directly calling the method?

File: spec/transcryptor/instance_spec.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          :attr_encrypted, :column_1,
+          key: 'column_1_key_qwe_qwe_qwe_qwe_qwe', mode: :per_attribute_iv
+        )
+      end
+
+      let(:attr_encrypted_config_after) do
+        InstanceSpec.send(
```
Comment (ronaldtse) [inline spec/transcryptor/instance_spec.rb]: Is it necessary to use send( here or just directly calling the method?

File: spec/transcryptor/instance_spec.rb (hunk 1 above)
Comment (DmitryDrobotov) [inline spec/transcryptor/instance_spec.rb]: Yes, @ronaldtse . Because it is protected method, so I can not just simply call it.

**Comment by DmitryDrobotov** [comment]
@ronaldtse could you help me to fix this typos? I am not that great in English, you know :)

**Pull request merged**